pkg/util/tencent/tencent.go

```diff
@@ -0,0 +1,101 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package tencent
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// declare these as vars not const to ease testing
+var (
+	metadataURL = "http://metadata.tencentyun.com"
```

Comment:
I am worried about this being a domain instead of a hardcoded IP like other cloud providers. IMO this has pretty serious security implication as anyone controlling the DNS response could force agents to make an HTTP request across the web.
@oiooj do you know if there is any way to use an IP instead? If not we would at least need to do the DNS query manually and check the IP is in a private range.

I am also worried about customers asking why the agent makes DNS requests to `http://metadata.tencentyun.com` as this is something they are often monitoring.

Reply:
@arbll Yes, we should use a IP, and this IP will not changed.